### PR TITLE
Fix coverage gaps in distance subfolder

### PR DESCRIPTION
**Pull request by Schwarf**
**Description:**
This PR adds additional unit tests to cover edge cases and address minor coverage gaps in the distance submodule.
Apart from the removal of some unused `#include` directives, no production code has been modified.

### CODE REVIEW

File: networkit/cpp/distance/test/DistanceGTest.cpp [hunk 1]
```diff
@@ -82,8 +82,49 @@ TEST_F(DistanceGTest, testVertexDiameterPedantically) {
...
+    Graph G(5);
+    std::vector<double> distanceHeu = {1.0, 2.0, 3.0, 4.0, 5.0};
+
+    node source = 0;
+    node target = 0;
+
+    AStar astar(G, distanceHeu, source, target);
+    astar.run();
+    double expected_result = 0.0;
+    EXPECT_DOUBLE_EQ(astar.getDistance(), expected_result);
```
Comment (angriman) [inline networkit/cpp/distance/test/DistanceGTest.cpp:95]: `EXPECT_DOUBLE_EQ(astar.getDistance(), 0.0);` is clear enough.

File: networkit/cpp/distance/test/DistanceGTest.cpp [hunk 2]
```diff
@@ -82,8 +82,49 @@ TEST_F(DistanceGTest, testVertexDiameterPedantically) {
...
+    EXPECT_THROW({ AStar astar(G, distanceHeu, source, invalid_target); }, std::runtime_error);
+}
+
+TEST_F(DistanceGTest, testAStarConstructorThrowsMismatchHeuristicAndNodeCount) {
+    Graph G(5);
+    std::vector<double> distanceHeu = {1.0, 2.0}; // size mismatch (2 instead of 5)
+    node source = 0;
+    node target = 1;
+
+    EXPECT_THROW({ AStar astar_bad(G, distanceHeu, source, target); }, std::runtime_error);
```
Comment (angriman) [inline networkit/cpp/distance/test/DistanceGTest.cpp:124]: `EXPECT_THROW({ AStar astar_bad(G, distanceHeu, 0, 1); }, std::runtime_error);`

File: networkit/cpp/distance/test/DistanceGTest.cpp [hunk 3]
```diff
@@ -152,6 +193,29 @@ TEST_F(DistanceGTest, testAStar) {
     testMesh(25, 5);
 }
 
+TEST_F(DistanceGTest, testAlgebraicDistanceContructorThrowsForInvalidOmegaValues) {
+    Graph G(5, true, true);
+    double invalid_omega_value1 = 1.1;
+    double invalid_omega_value2 = -0.3;
```
Comment (angriman) [inline networkit/cpp/distance/test/DistanceGTest.cpp:199]: You can inline these two in the constructor call as well.

File: networkit/cpp/distance/test/DistanceGTest.cpp [hunk 4]
```diff
@@ -328,6 +392,18 @@ TEST_F(DistanceGTest, testBidirectionalBFSTargetEqualSource) {
...
+TEST_F(DistanceGTest, testBidirectionalBFSGetPathIsEmpty) {
+    Graph G(3, false, false);
+    G.addEdge(0, 1);
+    G.addEdge(0, 2);
+
+    BidirectionalBFS bbfs(G, 1, 1, false);
+    bbfs.run();
+    auto result = bbfs.getPath();
+    std::vector<node> expected_result{};
+    EXPECT_EQ(result, expected_result);
```
Comment (angriman) [inline networkit/cpp/distance/test/DistanceGTest.cpp:404]: `EXPECT_THAT(bbfs.getPath(), testing::IsEmpty());`

File: networkit/cpp/distance/test/DistanceGTest.cpp [hunk 5]
```diff
@@ -682,6 +758,26 @@ TEST_P(DistanceGTest, testSPSPWithUnreachableTarget) {
             EXPECT_DOUBLE_EQ(apsp.getDistance(source, target), spsp.getDistance(source, target));
 }
 
+TEST_F(DistanceGTest, testMultiTargetBFSThrowsInvalidSourceWithTargetRange) {
+    Graph G(5);
+    node invalid_source = 6;
```
Comment (angriman) [inline networkit/cpp/distance/test/DistanceGTest.cpp:763]: You can inline this in the constructor call.

File: networkit/cpp/distance/test/DistanceGTest.cpp [hunk 6]
```diff
@@ -682,6 +758,26 @@ TEST_P(DistanceGTest, testSPSPWithUnreachableTarget) {
...
+    std::vector<node> targets = {0, 1, 2, 3};
+    EXPECT_THROW({ MultiTargetBFS astar(G, invalid_source, targets.begin(), targets.end()); },
+                 std::runtime_error);
+}
+
+TEST_F(DistanceGTest, testMultiTargetBFSThrowsWithOneInvalidTargetInTargetRange) {
+    Graph G(5);
+    node source = 0;
```
Comment (angriman) [inline networkit/cpp/distance/test/DistanceGTest.cpp:771]: Same here, just use `MultiTargetBFS astar(G, 0, ...);`

File: networkit/cpp/distance/test/SSSPGTest.cpp [hunk 7]
```diff
@@ -52,6 +51,36 @@ TEST_F(SSSPGTest, testDijkstra) {
         EXPECT_LE(sssp.distance(stack[i]), sssp.distance(stack[i + 1]));
 }
 
+TEST_F(SSSPGTest, testSSSPThrowsInvalidSource) {
+    constexpr int n = 5;
+    Graph G(n, true);
+    int source = 1;
+    BFS bfs(G, source);
+    int invalid_source = 6;
```
Comment (angriman) [inline networkit/cpp/distance/test/SSSPGTest.cpp:59]: All these `int` variables are not needed, just inline the values in the constructors or functions.

File: networkit/cpp/distance/test/SSSPGTest.cpp [hunk 8]
```diff
@@ -52,6 +51,36 @@ TEST_F(SSSPGTest, testDijkstra) {
...
+    EXPECT_THROW(bfs.setSource(invalid_source), std::runtime_error);
+}
+
+TEST_F(SSSPGTest, testSSSPThrowsInvalidTarget) {
+    constexpr int n = 5;
+    Graph G(n, true);
+    int source = 1;
+    BFS bfs(G, source);
+    int invalid_target = 6;
+    EXPECT_THROW(bfs.setTarget(invalid_target), std::runtime_error);
```
Comment (angriman) [inline networkit/cpp/distance/test/SSSPGTest.cpp:69]: Same as above.

File: networkit/cpp/distance/test/SSSPGTest.cpp [hunk 9]
```diff
@@ -52,6 +51,36 @@ TEST_F(SSSPGTest, testDijkstra) {
...
+TEST_F(SSSPGTest, testSSSPThrowsStorepathNotTrueCallPredecessor) {
+    constexpr int n = 3;
+    Graph G(n, true);
+    G.addEdge(0, 2);
+    G.addEdge(1, 2);
+    int source = 0;
+    BFS bfs(G, source, false);
+    bfs.setTarget(2);
+    bfs.run();
+    EXPECT_THROW(bfs.getPredecessors(1), std::runtime_error);
```
Comment (angriman) [inline networkit/cpp/distance/test/SSSPGTest.cpp:81]: Here as well.

**Review by angriman — CHANGES_REQUESTED** [review]
Thank you for improving coverage, I left a few minor comments.

**Comment by angriman** [comment]
In that case you can always add inline comments, for example:
```cpp
AStar astar(G, distanceHeu, /*source=*/0, /*target=*/6);
```

**Comment by Schwarf** [comment]
Hi @fabratu,
I’m seeing a build failure in the CI (Python/Cython error CYTHON_METH_FASTCALL undeclared) here:
https://github.com/networkit/networkit/actions/runs/15710850694/job/44268240777?pr=1326
I didn’t touch those files in my PR. Could you take a look and confirm whether this is a pre-existing issue? Thanks!

**Comment by fabratu** [comment]
Since all review comments concerned minor edits, this PR will now be merged.
